dart/common/Resource.hpp

```diff
@@ -75,6 +75,12 @@ class Resource
   /// \param[in] _count Number of elements, each of _size bytes.
   /// \note This method has the same API as the standard fread function.
   virtual std::size_t read(void *_buffer, std::size_t _size, std::size_t _count) = 0;
+
+  /// Reads all remaining data from this resource, and returns it as a string.
```

Comment:
I don't think "read remaining data" is true. I suspect that this function will fail if it is called on a `Resource` that has been partially consumed because `getSize()` returns the size of the _entire_ resource and `read` starts reading from the current cursor position.